src/core/crypto/singleKey.ts

```diff
@@ -0,0 +1,199 @@
+import { Deserializer, Serializable, Serializer } from "../../bcs";
+import {
+  AnyPublicKeyVariant,
+  AnySignatureVariant,
+  HexInput,
+  SigningScheme as AuthenticationKeyScheme,
+  SigningSchemeInput,
+} from "../../types";
+import { AuthenticationKey } from "../authenticationKey";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+
+type PublicKeyInput = Ed25519PublicKey | Secp256k1PublicKey;
+type SignatureInput = Ed25519Signature | Secp256k1Signature;
+
+/**
+ * Represents any public key supported by Aptos.
+ *
+ * Since [AIP-55](https://github.com/aptos-foundation/AIPs/pull/263) Aptos supports
+ * `Legacy` and `Unified` authentication keys.
+ *
+ * Any unified authentication key is represented in the SDK as `AnyPublicKey`.
+ */
+export class AnyPublicKey extends Serializable {
```

Comment:
file name is `singleKey` but class name is still `AnyPublicKey` - could we go with one name?